library/rsa.c

```diff
@@ -398,23 +398,24 @@ int mbedtls_rsa_export_raw( const mbedtls_rsa_context *ctx,
     return( ret );
 }
 
+int mbedtls_rsa_is_private( const mbedtls_rsa_context *ctx )
+{
+    return(
+        mbedtls_mpi_cmp_int( &ctx->N, 0 ) != 0 &&
+        mbedtls_mpi_cmp_int( &ctx->P, 0 ) != 0 &&
+        mbedtls_mpi_cmp_int( &ctx->Q, 0 ) != 0 &&
+        mbedtls_mpi_cmp_int( &ctx->D, 0 ) != 0 &&
+        mbedtls_mpi_cmp_int( &ctx->E, 0 ) != 0 );
```

Comment:
Two comments - first this seems to accept values which are negative, values which could be deserialized from the DER structure. Also it seems to ignore the CRT params, which should probably also be checked, if CRT is enabled at least.

Reply:
Thanks for your feedback! Actually I think the checks in `rsa_check_context()` (called from `mbedtls_rsa_complete()` called from `mbedtls_pk_parse_key()`) already prevent that, but I agree that it should be more obvious. I was intending to rework the PR anyway, as I'm no longer convinced `mbedtls_rsa_is_private()` is the right function here, so I'll keep that in mind when reworking.